rockhound/bedmap2.py

```diff
@@ -0,0 +1,122 @@
+"""
+Load the Bedmap2 datasets for Antarctica.
+"""
+import os
+import tempfile
+from zipfile import ZipFile
+
+import xarray as xr
+
+from .registry import REGISTRY
+
+DATASETS = [
+    "bed",
+    "surface",
+    "thickness",
+    "icemask_grounded_and_shelves",
+    "rockmask",
+    "lakemask_vostok",
+    "grounded_bed_uncertainty",
+    "thickness_uncertainty_5km",
+    "coverage",
+    "geoid",
+]
+
+
+def fetch_bedmap2(datasets, load=True):
+    """
+    Fetch the Bedmap2 datasets for Antarctica.
+
+    Bedmap2 is a suite of gridded products describing surface elevation,
+    ice-thickness, the sea ﬂoor and subglacial bed elevation of the Antarctic south
+    of 60°S [BEDMAP2]_.
+    The datasets are downloaded as ``tiff`` files and loaded into a
+    :class:`xarray.Dataset` object.
+
+    Each dataset is projected in Antarctic Polar Stereographic projection, latitude of
+    true scale -71 degrees south, datum WGS84. All heights are in metres relative to
+    sea level as defined by the g104c geoid.
+
+    The available datasets are:
+
+    - ``bed``: bedrock height
+    - ``surface``: ice surface height
+    - ``thickness``: ice thickness
+    - ``icemask_grounded_and_shelves``: mask showing the grounding line and the extent
+      of the floating ice shelves
+    - ``rockmask``: mask showing rock outcrops
+    - ``lakemask_vostok``: mask showing the extent of the lake cavity of Lake Vostok
+    - ``grounded_bed_uncertainty``: ice bed uncertainty grid
+    - ``thickness_uncertainty_5km``: ice thickness uncertainty grid
+    - ``coverage``: binary grid showing the distribution of ice thickness data used
+      in the grid of ice thickness
+    - ``geoid``: values to convert from heights relative to WGS84 datum to heights
+      relative to EIGEN-GL04C geoid (to convert back to WGS84, add this grid)
+
+    .. warning ::
+        Loading a great number of datasets may require a fair amount of memory that
+        could crash your system. We recommend loading only the needed datasets.
```

Comment:
Now that I read this docstring it seems kind of contradictory to say this and then offer an `all` option. It might be better to remove the `all`. It'll make the code simpler and we don't have to test loading all the data, which will certainly crash the CI.